### PR TITLE
Only auto-advance if your turn

### PR DESCRIPTION
**Pull request by BHydden**
PR to resolve issue #892 

WIP - LF feedback @anoek 

I get a Type Error preventing the site from loading when I try to communicate with src/components/Notifications/Notifications.tsx from src/lib/goban.ts

https://i.imgur.com/Vc9mkqJ.png

Looks like some kind of compiling dependency issue of things loading in the wrong order, could you point me in the right direction? I was thinking of moving the whole autoadvance function from lib/goban over to Notifications... is that the right idea or will that not resolve this dependency issue / create new issues?

### CODE REVIEW

**Comment by BHydden** [comment]
I can't believe it passes the shippable test when the code breaks the whole site 🤣

**Comment by anoek** [comment]
Yeah that's a dependency problem

You could move that logic to https://github.com/online-go/online-go.com/blob/devel/src/views/Game/Game.tsx#L464 where we listen for that event maybe?

**Comment by BHydden** [comment]
Ok thanks, I'll give that a go

File: src/lib/goban.ts [hunk 1]
```diff
@@ -106,20 +105,6 @@ export class Goban extends OGSGoban {
...
-        if (!user.anonymous && /^\/game\//.test(this.getLocation())) {
-            /* if we just moved */
-            if (this.engine.playerNotToMove() === user.id) {
-                if (!isLiveGame(this.engine.time_control) && preferences.get("auto-advance-after-submit")) {
-                    this.emit("advance-to-next-board");
-                }
-            }
-        }
-    }
-}
```
Comment (BHydden) [inline src/lib/goban.ts]: Deleted one too many }

**Comment by BHydden** [comment]
Property 'engine' does not exist on type 'Game'.

**Comment by BHydden** [comment]
Pretty sure this is good to go @anoek 😃 thanks for your help

**Comment by anoek** [comment]
Thanks!

**Pull request merged**